### PR TITLE
Add Plural fathead

### PR DESCRIPTION
**Pull request by wilkox**
This is an attempt to convert the submitted plural Goodie ([see this pull request](https://github.com/duckduckgo/zeroclickinfo-goodies/pull/493)) to a Fathead. I've pasted the original pull request description below for context.

A few notes/problems:

- I can't figure out how to specify triggers or any post-trigger processing (i.e. what would be performed by `handle remainder =>` in a Goodie). I've left the Goodie's `triggers` and `handle` code in `Plural.pm` for reference, but presumably this will need to be removed or rewritten for the final version.
- Since the original Goodie output css-styled HTML, I've written the parse script to output fully-styled HTML in the 'abstract' field of `output.txt`. Is this acceptable for a Fathead?
- There didn't seem to be any existing test files for Fatheads, so I've created a `t/` directory.
- I've left in the JSON code as a reference for how the original Goodie worked; I'll remove this once the Fathead is working.

Any help would be great!

**What does your instant answer do?**
Returns the plural form(s) of an English noun.

**What problem does your instant answer solve (Why is it better than organic links)?**
Saves the user from navigating to a dictionary page and searching for the plural form.

**What is the data source for your instant answer? (Provide a link if possible)**
Wiktionary: parsed from [this dump file](http://dumps.wikimedia.org/enwiktionary/20140522/enwiktionary-20140522-pages-articles.xml.bz2)

**Why did you choose this data source?**
It's comprehensive, accurate and free

**Are there any other alternative (better) data sources?**
Not aware of any better source for free.

**What are some example queries that trigger this instant answer?**
"plural starfish"
"how do you pluralize supernova"

**Which communities will this instant answer be especially useful for? (gamers, book lovers, etc)**
Writers, students.

**Is this instant answer connected to a DuckDuckHack [instant answer idea](https://duck.co/ideas)?**
Yes, [this one](https://duck.co/ideas/idea/802/pluralize-english-words)

**Which existing instant answers will this one supercede/overlap with?**
As the user could potentially request the plural form of any noun, it would be possible to simultaneously trigger any other instant answer that triggers on a noun.

**Are you having any problems? Do you need our help with anything?**
See above.

**\*\*Note:** Please attach a screenshot for new instant answer pull requests, and for pull requests which modify the look/design of existing instant answers.

image

## Checklist
Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information
[X] Wrote test file and added to t/ directory
[Xish] Verified that instant answer adheres to design guidelines(https://github.com/duckduckgo/duckduckgo-documentation/blob/master/duckduckhack/styleguides/design_styleguide.md)
[X] Tested cross-browser compatability

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome   
        [] Firefox         
        [] Opera           
        [] IE 10           

    - Windows 7
        [] Google Chrome   
        [] Firefox         
        [] Opera           
        [] IE 8            
        [] IE 9            
        [] IE 10           

    - Windows XP
        [] IE 7            
        [] IE 8            

    - Mac OSX
        [X] Google Chrome   
        [X] Firefox         
        [] Opera           
        [X] Safari          

    - iOS (iPhone)
        [] Safari Mobile   
        [] Google Chrome   
        [] Opera           

    - iOS (iPad)
        [] Safari Mobile   
        [] Google Chrome   
        [] Opera            

    - Android
        [] Firefox         
        [] Native Browser  
        [] Google Chrome   
        [] Opera
```
https://duck.co/ia/view/plural

### CODE REVIEW

**Comment by moollaza** [comment]
ping! @wilkox what's the status on this? Do we need to take another look at it?

**Comment by wilkox** [comment]
Yep it is ready for another review. The main change in my last commit (b35e45) was to put the entire trigger string in the first column of output.txt

**Comment by jdorweiler** [comment]
@wilkox I'm getting this error. 
```
Feature bundle "5.18" is not supported by Perl 5.16.3 at /usr/local/ddg.cpan/perl5/lib/perl5/Modern/Perl.pm line 40.
BEGIN failed--compilation aborted at extract_plurals.pl line 3.

```

**Comment by wilkox** [comment]
@jdorweiler sorry about that, 838660a should fix it.

**Comment by javathunderman** [comment]
@moollaza ping

**Comment by jdorweiler** [comment]
@javathunderman Thanks, I'm checking it out now.
@wilkox could you update your branch?  We moved the fathead fetch/parse scripts into `/share/fathead/`

**Comment by wilkox** [comment]
@jdorweiler no worries, updated and moved the scripts.

File: share/fathead/plural/extract_plurals.pl [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+use Lingua::EN::Inflect qw(WORDLIST);
+use utf8;
+$|++;
+
+#Regex to catch weirdness (malformed tags etc.)
+# we don't want leaking into the results
+my $unsanitary = qr/[^\p{L}\s\-']/;
+
+#Define triggers
+my @triggersStart = ("plural of", "pluralise", "pluralize", "what is the plural of");
```
Comment (jdorweiler) [inline share/fathead/plural/extract_plurals.pl]: You can skip these and just have one entry per word.  I'll add these as trigger words on the backend.

File: share/fathead/plural/extract_plurals.pl [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+my @triggersEnd = ("plural");
```
Comment (jdorweiler) [inline share/fathead/plural/extract_plurals.pl]: Same here.  I'll add this as a trigger.

**Comment by wilkox** [comment]
@jdorweiler I've removed the hard-coded triggers; output now has one entry per word.

**Comment by javathunderman** [comment]
ping @jdorweiler @wilkox Any updates on this?

**Comment by jdorweiler** [comment]
@javathunderman @wilkox sorry for the delay.  I put this on a test server https://ddh10.duckduckgo.com/?q=plural+aabp&ia=about
Just waiting on some internal feed back on it.

**Comment by javathunderman** [comment]
@jdorweiler No problem.

**Comment by jdorweiler** [comment]
@wilkox @javathunderman This looks good, thanks!  I'll merge but it might be a week or two before we can deploy it.  We're working out a new fathead deploy process.  I'll update the pr when it's live.

**Comment by javathunderman** [comment]
:+1:

**Comment by wilkox** [comment]
Thanks everyone!

**Comment by jdorweiler** [comment]
@wilkox Finally live! https://duckduckgo.com/?q=plural+octopus&ia=about

**Pull request merged**